nimare/decode/base.py

```diff
@@ -92,7 +92,6 @@ def fit(self, dataset, drop_invalid=True):
             Whether to automatically ignore any studies without the required data or not.
             Default is True.
 
-
         Returns
         -------
         :obj:`~nimare.results.MetaResult`
```

Comment:
Hmmm, if we say that's what we return then we probably should.

How much work do you think that would be?

Reply:
I think it will be easy to implement. We will only need to add:
```python
return MetaResult(self, mask=masker, maps=maps, tables=tables, description=description)
```
where `maps` is a dictionary with `features_` as the keys and `images_` as the values.
We could even add a `description` for the boilerplate in the future.

Reply:
Although, it seems like no other decoders currently return anything using `fit`, so could also just decide to change the docstring of `fit` on the base `Decoder` class. 

In sklearn, `fit` usually is an inplace operation, and `transform` is the one that returns a result

Reply:
> Although, it seems like no other decoders currently return anything using fit, so could also just decide to change the docstring of fit on the base Decoder class.

Yeah, it seems like the discrete decoders are not fitted to a dataset. Returning a MetaResult object would be only supported for continuous decoders. 

> In sklearn, fit usually is an inplace operation, and transform is the one that returns a result

I think in nimare `fit` currently works in-place for `Decoder`, but it can also returns a `MetaResult` object for `Estimator`:
https://github.com/neurostuff/NiMARE/blob/4b14bcb94c799f637981b40868b3522057c99729/nimare/estimator.py#L95-L109

Reply:
How would transform work? Would it take the `MetaResult` as input?

Reply:
I think the `transform` would still take the image to be decoded as input. The `MetaResult` as input would indicate that the MetaResult is being transformed, as we have in `Corrector`. Perhaps if the `out_df` is added to the tables MetaResult object, it could justify being an input in transform; and then transform will return the new MetaResult object.
`fit()` will still populate the attributes (masker, images_, and features_) used by `transform` in place. Returning a MetaResult will then allow users to easily save the maps, tables, and boilerplate.

Reply:
Ah, I see. I think that makes sense, actually.

How do we manage loading a `MetaResult` object into the `Decoder` object's internals?

Reply:
One option: make the result object an optional argument to `transform` which is required if `fit` has not been called

Reply:
> How do we manage loading a MetaResult object into the Decoder object's internals?

I think it will be created and returned by `fit()` and by the new method `load_imgs`. Internally it would be used by  `transform` if we decide to add the result table to it. If that is the case, we could just pass it as an argument and return it with the result table.

Reply:
That sounds good to me